router/handle_lifecycle.go

```diff
@@ -347,7 +347,10 @@ func (rt *Handle) Shutdown() {
 	rt.logger.Infof("Shutting down router: %s", rt.destType)
 	rt.backgroundCancel()
 
-	<-rt.startEnded     // wait for all workers to stop first
+	<-rt.startEnded // wait for all workers to stop first
+	if rt.throttlerFactory != nil {
+		rt.throttlerFactory.Shutdown()
+	}
```

Comment:
If there is no scenario where throttler factory is nil
```suggestion
	rt.throttlerFactory.Shutdown()
```

Reply:
Removed nil check